### PR TITLE
OpenCTI SOAR playbooks

### PR DESCRIPTION
**Pull request by anki-narravula**
Required items, please complete
   
   Change(s):
   - Added new solution OpenCTI, it contains Playbooks

   Reason for Change(s):
   - New addition

   Version Updated:
   - 1.0.0 Initial version

   Testing Completed:

image

image

image

image

image

image

### CODE REVIEW

File: Solutions/OpenCTI/Playbooks/OpenCTICustomConnector/readme.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+| **Run GraphQL query** | Run any supported GraphQL query and get the response in json format |
+
+
+### Deployment instructions 
+1. Deploy the Custom Connector by clicking on "Deploy to Azure" button. This will take you to deplyoing an ARM Template wizard.
+2. Fill in the required paramteres:
+    * Custom Connector Name : Enter the Custom connector name (ex:OpenCTICustomConnector)
+    * OpenCTI Host URL: Enter the OpenCTI URL or Host (ex: demo.opencti.io)
+
+[![Deploy to Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2FSolutions%2F%2OpenCTI%2FPlaybooks%2FOpenCTICustomConnector%2Fazuredeploy.json)
```
Comment (manishkumar1991) [inline Solutions/OpenCTI/Playbooks/OpenCTICustomConnector/readme.md:26]: Here in URL "Master" is not present like in other readme.md files its present : for example : 

(https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2F**master**%2FSolutions%2FOpenCTI%2FPlaybooks%2FOpenCTIPlaybooks%2FOpenCTI-CreateIndicator%2Fazuredeploy.json)

File: Solutions/OpenCTI/Playbooks/OpenCTIPlaybooks/OpenCTI-GetIndicatorsStream/readme.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    
+### Post-Deployment instructions 
+#### a. Authorize connections
+Once deployment is complete, you will need to authorize each connection.
+1.	Click the Azure Sentinel connection resource
+2.	Click edit API connection
+3.	Click Authorize
+4.	Sign in
+5.	Click Save
+6.	Repeat steps for Okta Api  Connection (For authorizing the Okta API connection, API Key needs to be provided)
```
Comment (manishkumar1991) [inline Solutions/OpenCTI/Playbooks/OpenCTIPlaybooks/OpenCTI-GetIndicatorsStream/readme.md:31]: Okta is mentioned in all readme files

File: Solutions/OpenCTI/Playbooks/OpenCTIPlaybooks/OpenCTI-CreateIndicator/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,2775 @@
...
+    "contentVersion":  "1.0.0.0",
+        "metadata":  {
+        "title":  "Update Indicator in OpenCTI",
+        "description":  "This playbook creates indicator in OpenCTI based on the entities info present in Sentinel incident. This playbook search in OpenCTI for indicatoes based on the entities (Account, Host, IP, FileHash, URL) present in Microsoft Sentinel incident. If it presnts in OpenCTI, updates the indicator confidence (also information will be added to incident comment) otherwise it creates new indicator in OpenCTI",
+        "prerequisites":  ["1. OpenCTI Custom Connector needs to be deployed prior to the deployment of this playbook under the same subscription.",
+        "2. API key. To get API Key, login into your OpenCTI instance dashboard and navigate to User profile page --> API Access."],
+        "postDeployment":  ["1. Once deployment is complete, you will need to authorize each connection.",
+        "2. Navigate to playbook --> API Connections --> Select connections one by one --> Edit API Connection --> (if required) Enter API key or credentials --> Save"],
+        "prerequisitesDeployTemplateFile":  "../../OktaCustomConnector/azuredeploy.json",
```
Comment (manishkumar1991) [inline Solutions/OpenCTI/Playbooks/OpenCTIPlaybooks/OpenCTI-CreateIndicator/azuredeploy.json:11]: Okta mentioned here

File: Solutions/OpenCTI/Playbooks/OpenCTIPlaybooks/OpenCTI-GetIndicatorsStream/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,790 @@
...
+    "contentVersion":  "1.0.0.0",
+    "metadata":  {
+        "title":  "Get Indicators into Sentinel from OpenCTI",
+        "description":  "This playbook fetches indicators from OpenCTI and send to Sentinel. Supported types are Domain, File, IPv4, IPv6, Account, Url.  This runs for every 10 minutes",
+        "prerequisites":  ["1. OpenCTI Custom Connector needs to be deployed prior to the deployment of this playbook under the same subscription.",
+        "2. API key. To get API Key, login into your OpenCTI instance dashboard and navigate to User profile page --> API Access.",
+        "3. OpenCTI-ImportToSentinel must be installed and keep playbook name and batchname handy"],
+        "postDeployment":  ["1. Once deployment is complete, you will need to authorize each connection.",
+        "2. Navigate to playbook --> API Connections --> Select connections one by one --> Edit API Connection --> (if required) Enter API key or credentials --> Save"],
+        "prerequisitesDeployTemplateFile":  "../../OktaCustomConnector/azuredeploy.json",
```
Comment (manishkumar1991) [inline Solutions/OpenCTI/Playbooks/OpenCTIPlaybooks/OpenCTI-GetIndicatorsStream/azuredeploy.json:12]: Please check for okta word in all playbook. and OpenCTI-GetIndicatorsStream playboook expects Import-to-sentinel to be pre deployed other wise this playbook deployment fails , so please add that as well in prerequisite

Comment (anki-narravula) [reply Solutions/OpenCTI/Playbooks/OpenCTIPlaybooks/OpenCTI-GetIndicatorsStream/azuredeploy.json:12]: yeah right, currently more than 1 dependency is not supported, so I just mentioned OpenCTICustomConnector now. Other dependency, I have written in prerequisites instructions

File: Solutions/OpenCTI/Playbooks/OpenCTICustomConnector/readme.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+### Authentication methods this connector supports
+
+*  API Key authentication
+
+### Prerequisites for deploying Custom Connector
+1. OpenCTI host end point or url should be known(ex : demo.opencti.io)
+2. API key. To get API Key, login into your OpenCTI instance dashboard and navigate to User profile page --> API Access.
+
+
+## Actions supported by okta custom connector
```
Comment (manishkumar1991) [inline Solutions/OpenCTI/Playbooks/OpenCTICustomConnector/readme.md:13]: Okta is mentioned here , please correct

File: Solutions/OpenCTI/Playbooks/OpenCTIPlaybooks/OpenCTI-CreateIndicator/readme.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    
+### Post-Deployment instructions 
+#### a. Authorize connections
+Once deployment is complete, you will need to authorize each connection.
+1.	Click the Azure Sentinel connection resource
+2.	Click edit API connection
+3.	Click Authorize
+4.	Sign in
+5.	Click Save
+6.	Repeat steps for Okta Api  Connection (For authorizing the Okta API connection, API Key needs to be provided)
```
Comment (manishkumar1991) [inline Solutions/OpenCTI/Playbooks/OpenCTIPlaybooks/OpenCTI-CreateIndicator/readme.md:33]: Okta is mentioned here , please correct

File: Solutions/OpenCTI/Playbooks/OpenCTIPlaybooks/OpenCTI-EnrichIncident/readme.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    
+### Post-Deployment instructions 
+#### a. Authorize connections
+Once deployment is complete, you will need to authorize each connection.
+1.	Click the Azure Sentinel connection resource
+2.	Click edit API connection
+3.	Click Authorize
+4.	Sign in
+5.	Click Save
+6.	Repeat steps for Okta Api  Connection (For authorizing the Okta API connection, API Key needs to be provided)
```
Comment (manishkumar1991) [inline Solutions/OpenCTI/Playbooks/OpenCTIPlaybooks/OpenCTI-EnrichIncident/readme.md:33]: Okta is mentioned here , please correct

File: Solutions/OpenCTI/Playbooks/OpenCTIPlaybooks/OpenCTI-GetIndicatorsStream/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,798 @@
...
+    "contentVersion":  "1.0.0.0",
+    "metadata":  {
+        "title":  "Get Indicators into Sentinel from OpenCTI",
+        "description":  "This playbook fetches indicators from OpenCTI and send to Sentinel. Supported types are Domain, File, IPv4, IPv6, Account, Url.  This runs for every 10 minutes",
+        "prerequisites":  ["1. OpenCTI Custom Connector needs to be deployed prior to the deployment of this playbook under the same subscription.",
+        "2. API key. To get API Key, login into your OpenCTI instance dashboard and navigate to User profile page --> API Access.",
+        "3. OpenCTI-ImportToSentinel must be installed and keep playbook name and batchname handy"],
+        "postDeployment":  ["1. Once deployment is complete, you will need to authorize each connection.",
+        "2. Navigate to playbook --> API Connections --> Select connections one by one --> Edit API Connection --> (if required) Enter API key or credentials --> Save"],
+        "prerequisitesDeployTemplateFile":  "../../OpenCTICustomConnector/azuredeploy.json",
```
Comment (manishkumar1991) [inline Solutions/OpenCTI/Playbooks/OpenCTIPlaybooks/OpenCTI-GetIndicatorsStream/azuredeploy.json:12]: Import to sentinel should be there as well before installing this play book , else playbook deployment of Getindicator stream will fail

Comment (anki-narravula) [reply Solutions/OpenCTI/Playbooks/OpenCTIPlaybooks/OpenCTI-GetIndicatorsStream/azuredeploy.json:12]: yes, its mentioned in Pre-deployment steps

**Review by manishkumar1991 — APPROVED** [review]
approved

**Pull request merged**